### PR TITLE
131 dev build changes

### PR DESCRIPTION
**Pull request by Toshbrown**
Addresses #131

Add a new script called data **databox-install-component** that:

- Takes a component name as an option (me-box/app-os-monitor)
- Pulls the required source code 
- Builds the image 
- Adds the Databox manifest to the local app store

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,28 +55,35 @@ UI, ready for you to install.
...
 ```
 
-Databox has a number of platform components, divided into two parts:  Core and User components.
+When you start in development mode only the `core-components` are built from source. If you wish to develop one of the available apps or drivers then you can add them to you local install using:
+
+```
+./databox-install-component driver-os-monitor
+```
+
+This will download and build the code on your machine and upload the Databox manifest to your local app store. You can also use this with your repositories  and forks using:
```
Comment (mor1) [inline README.md:72]: Is a local app store guaranteed to be running?

File: README.md [hunk 2]
```diff
@@ -55,28 +55,35 @@ UI, ready for you to install.
...
+* [Databox container manager](https://github.com/me-box/core-container-manager)
 * [databox-arbiter](https://github.com/me-box/core-arbiter)
-
 * [databox-export-service](https://github.com/me-box/core-export-service)
-
 * [databox-store-json](https://github.com/me-box/store-json)
-
 * [databox-store-timeseries](https://github.com/me-box/store-timeseries)
-
 * [databox-app-server](https://github.com/me-box/platform-app-server) Server for storing and serving databox manifests
```
Comment (mor1) [inline README.md:88]: Worth adding short descriptions (and consistent naming format) to all of these

File: README.md [hunk 3]
```diff
@@ -86,16 +93,16 @@ Databox has a number of platform components, divided into two parts:  Core and U
...
-#### Apps 
+* [driver-tplink-smart-plug](https://github.com/me-box/driver-tplink-smart-plug)
+#### Apps
 * [app-light-graph](https://github.com/me-box/app-light-graph)
 * [app-twitter-sentiment](https://github.com/me-box/app-twitter-sentiment)
 * [app-os-monitor](https://github.com/me-box/app-os-monitor)
-                
+
 ### Libraries for writing drivers and apps
 For writing a new driver or app for Databox, one needs [Databox Api's](./documents/api_specification.md). To make app/driver development easy, we have wrapped Databox api's in [nodejs](https://nodejs.org/en/), [python](https://docs.python.org/3.4/library/index.html) and [go](https://golang.org/). Using any of these libraries, a developer can build their databox app/driver.
```
Comment (mor1) [inline README.md]: s/[aA]pi/API/g

File: README.md [hunk 4]
```diff
@@ -86,16 +93,16 @@ Databox has a number of platform components, divided into two parts:  Core and U
...
-* [lib-node-databox](https://github.com/me-box/node-databox): Databox nodejs api library for building databox apps and drivers. 
-* [lib-python-databox](https://github.com/me-box/lib-python-databox): Databox python api library for building databox apps and drivers. 
+* [lib-node-databox](https://github.com/me-box/node-databox): Databox nodejs api library for building databox apps and drivers.
+* [lib-python-databox](https://github.com/me-box/lib-python-databox): Databox python api library for building databox apps and drivers.
 * [lib-go-databox](https://github.com/me-box/lib-go-databox): Databox go api library for building databox apps and drivers.
```
Comment (mor1) [inline README.md]: s/api/API/g

python -> Python etc

File: databox-install-component [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+       `"see https://git-scm.com/"
+fi
+
+if ! [ -x "$(command -v curl)" ]; then
+  die 1 "curl is not installed;"`
+       `"https://curl.haxx.se"
+fi
+
+#include helper functions
+source ./scripts/utils
```
Comment (mor1) [inline databox-install-component]: source things first? (else where does `die` come from?)

File: databox-install-component [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+function test_assert {
+  if [ "$1" != "$2" ]
+  then
+    fail "$3" "$1"
+  else
+    success "$3"
+  fi
+}
+
```
Comment (mor1) [inline databox-install-component]: the above three might all move to `scripts/utils` perhaps?
(and I think convention would be that "library script files" get a `.sh` extension; commands don't)

File: databox-install-component [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+usage() {
+  echo ""
+  echo "Please invoke this script with the component you would like to install."
+  echo "Usage: databox-pin [OPTION...] [GITHUB-USER/COMPONENT-NAME]"
```
Comment (mor1) [inline databox-install-component]: `databox-pin`?

**Review by mor1 — COMMENTED** [review]
Minor tweaks/corrections suggested
Progress though :)

File: databox-install-component (hunk 7 above)
Comment (haddadi) [reply databox-install-component]: @mor1 see  https://github.com/me-box/databox/pull/99

File: README.md (hunk 1 above)
Comment (Toshbrown) [reply README.md:72]: For now, it is run in all modes and the container manager UI will query both the local and remote app store. If an image is present in both then a drop-down list is displayed to select the one you want.

File: README.md (hunk 2 above)
Comment (Toshbrown) [reply README.md:88]: Should this be a separate issue for documentation? Rather than part of this PR

Comment (pooyadav) [reply README.md:88]: I'll create a new PR for this.

File: README.md (hunk 3 above)
Comment (pooyadav) [reply README.md]: I didn't get this comment.

Comment (yousefamar) [reply README.md]: (Regex for replace instances of Api and api with API globally)

Comment (pooyadav) [reply README.md]: Got it :)

Comment (Toshbrown) [reply README.md]: I've fixed this one

File: README.md (hunk 4 above)
Comment (Toshbrown) [reply README.md]: I've fixed this one

File: README.md (hunk 2 above)
Comment (Toshbrown) [reply README.md:88]: Cheers 👍

**Comment by Toshbrown** [comment]
ping @mor1 RTM?

File: README.md (hunk 1 above)
Comment (mor1) [reply README.md:72]: Ok-- so the answer is "yes" currently? Will do for now, obvs we want to move away from that at some point.

File: README.md [hunk 8]
```diff
@@ -86,24 +93,24 @@ Databox has a number of platform components, divided into two parts:  Core and U
...
+
 ### Libraries for writing drivers and apps
-For writing a new driver or app for Databox, one needs [Databox Api's](./documents/api_specification.md). To make app/driver development easy, we have wrapped Databox api's in [nodejs](https://nodejs.org/en/), [python](https://docs.python.org/3.4/library/index.html) and [go](https://golang.org/). Using any of these libraries, a developer can build their databox app/driver.
-* [lib-node-databox](https://github.com/me-box/node-databox): Databox nodejs api library for building databox apps and drivers. 
-* [lib-python-databox](https://github.com/me-box/lib-python-databox): Databox python api library for building databox apps and drivers. 
-* [lib-go-databox](https://github.com/me-box/lib-go-databox): Databox go api library for building databox apps and drivers.
+For writing a new driver or app for Databox, one needs [Databox API's](./documents/api_specification.md). To make app/driver development easy, we have wrapped Databox API's in [nodejs](https://nodejs.org/en/), [python](https://docs.python.org/3.4/library/index.html) and [go](https://golang.org/). Using any of these libraries, a developer can build their databox app/driver.
```
Comment (mor1) [inline README.md:103]: (`API's` should be `APIs` -- plural needs no apostrophe...)

**Comment by mor1** [comment]
Merged on the basis that it's an improvement even if not perfect :)
Will await follow up PRs with other tweaks/fixes/improvements.

**Pull request merged**